examples/test-exec-doc.yml

```diff
@@ -0,0 +1,221 @@
+    - name: Run tests and handle PR
+      if: steps.check_changes.outputs.changed == 'true'
+      env:
+        GITHUB_TOKEN: ${{ secrets.PAT }}  # Personal Access Token from secrets
+        CHANGED_MARKDOWN_FILES: ${{ steps.check_changes.outputs.changed_files }}  # List of changed Markdown files
+      run: |
+        python <<EOF
+        import os
+        import subprocess
+        import shutil
+        import github
+
+        # Get GitHub token from environment variable
+        GITHUB_TOKEN = os.getenv('GITHUB_TOKEN')
+        # Authenticate with GitHub using PyGithub
+        g = github.Github(GITHUB_TOKEN)
+
+        # Check if 'ie' (Innovation Engine) is installed, install if not
+        if not shutil.which("ie"):
+            # Update package lists and install unzip
+            subprocess.run("sudo apt-get update && sudo apt-get install -y unzip", shell=True, check=True)
+
+            # Install Innovation Engine CLI
+            INNOVATION_ENGINE_VERSION = "v0.2.3"
+            subprocess.run(f"curl -Lks https://aka.ms/install-ie | /bin/bash -s {INNOVATION_ENGINE_VERSION}", shell=True, check=True)
+
+        # Retrieve repository information
+        result = subprocess.run(['git', 'config', '--get', 'remote.origin.url'], capture_output=True, text=True)
+        repo_url = result.stdout.strip()
+        parts = repo_url.split('/')
+        owner = parts[-2]                     # Repository owner
+        repo_name = parts[-1].replace('.git', '') # Repository name
+        # Access the repository using GitHub API
+        repo = g.get_repo(f"{owner}/{repo_name}")
+
+        # Get branch name from environment variable
+        BRANCH_NAME = os.environ['GITHUB_REF'].split('/')[-1]
+        if BRANCH_NAME != "main":
+            # Prepare to create or update a pull request to 'main'
+            PR_TITLE = f"Merge Exec Doc from {BRANCH_NAME} to main"
+            PR_BODY = "This PR attempts to publish your Exec Doc. Running tests on it now using Innovation Engine..."
+
+            # Check if a PR with the same title already exists
+            existing_pulls = repo.get_pulls(state='open', head=f"{owner}:{BRANCH_NAME}", base='main')
+            pr = None
+            for pull in existing_pulls:
+                if pull.title == PR_TITLE:
+                    pr = pull
+                    pr.edit(body=PR_BODY) # Update PR body if it exists
+                    break
+            if pr is None:
+                # Create a new pull request if none exists
+                pr = repo.create_pull(title=PR_TITLE, body=PR_BODY, head=f"{owner}:{BRANCH_NAME}", base='main')
+            pr_number = pr.number
+        else:
+            pr_number = None  # No PR needed if on 'main' branch
+
+        # Get the list of changed Markdown files
+        changed_files_env = os.getenv('CHANGED_MARKDOWN_FILES', '')
+        changed_files = [f.strip() for f in changed_files_env.split('\n') if f.strip().endswith('.md')]
+
+        if not changed_files:
```

Comment:
If there are no changed files, I don't think the pipeline should reach this point, right?

Reply:
good point - updated